tests/amm_contracts/utils.rs

```diff
@@ -149,3 +119,33 @@ pub fn execute_entry_point(
         &tx_execution_context,
     )
 }
+
+pub fn deploy(
+    state: &mut CachedState<InMemoryStateReader>,
+    path: &str,
+    config: &StarknetGeneralConfig,
+) -> (Address, [u8; 32]) {
+    let path = PathBuf::from(path);
+    let contract_class = ContractClass::try_from(path).unwrap();
+
+    let internal_deploy = InternalDeploy::new(
+        Address(0.into()),
+        contract_class.clone(),
+        vec![],
+        0.into(),
+        0,
+    )
+    .unwrap();
+    let class_hash = internal_deploy.class_hash();
+    state
+        .set_contract_class(&class_hash, &contract_class)
+        .unwrap();
+
+    let tx_execution_info = internal_deploy.apply(state, config).unwrap();
+
+    let call_info = tx_execution_info.call_info.unwrap();
+    let contract_address = call_info.contract_address;
+    let class_hash = call_info.class_hash.unwrap();
+
+    (contract_address, class_hash)
+}
```

Comment:
if this is a helper function for the test, i would declared it inside the test module. If it is set here then it can be used despise that is not the intended behavior.

Reply:
The issue with that is that in ERC721 we use the same function. That's why we use it in a utils.rs. Is there any alternative that would fit the criteria you mentioned to limit its use?